skia-safe/src/core/surface.rs

```diff
@@ -519,3 +518,31 @@ fn test_raster_direct() {
     let paint = Paint::default();
     surface.canvas().draw_circle((10, 10), 10.0, &paint);
 }
+
+#[test]
+fn test_drawing_as_mut_ergonomics() {
+    let mut surface = Surface::new_raster_n32_premul((16, 16)).unwrap();
+
+    // option1:
+    // - An _owned_ canvas can be drawn by transferring ownership.
+    {
+        let canvas = Canvas::new(ISize::new(16, 16), None).unwrap();
+        surface.draw(canvas, (10.0, 10.0), None);
+    }
+
+    // option2:
+    // - An &mut canvas can be drawn.
+
+    {
+        let mut canvas = Canvas::new(ISize::new(16, 16), None).unwrap();
+        surface.draw(&mut canvas, (10.0, 10.0), None);
+    }
+
+    // option3:
+    // - A canvas from another surface can be drawn.
+    {
+        let mut surface2 = Surface::new_raster_n32_premul((16, 16)).unwrap();

```

Comment:
The version I'm struggling with (I think) is like this:

``` Rust
fn test_borrowed(canvas: &mut Canvas) {
    let mut surface = Surface::new_raster_n32_premul((16, 16)).unwrap();
    surface.draw(canvas, (5.0, 5.0), None);
    surface.draw(canvas, (10.0, 10.0), None);
}
```

I'm still learning Rust but my conception is that this function should borrow the canvas mutably, and then release the mutable borrow at the end of the function call. Then in the next draw it will mutably borrow the canvas again. Unfortunately instead I get an error: "move occurs because `canvas` has type `&mut skulpin::skia_safe::Canvas`, which does not implement the `Copy` trait"

Reply:
Still learning, too. 

You are right, this undermines what I had intended originally. It seems that the conversion to the `impl AsMut<>` consumes the reference.